### PR TITLE
Refactor/replace black flake8 with ruff

### PR DESCRIPTION
**Pull request by sean-sinclair**
## Issue
https://github.com/equinor/fmu-sumo/issues/358

## Short description
- [x] Enable Linting and formatting
- [x] Perform Linting and formatting 
- [x] Create workflow for linting and formatting
- [x] Set up pre-commit with hooks

## Pre-review checklist
- [x] Read through all changes. No redundant `print()` statements, commented-out code, or other remnants from the development. 👀
- [x] Make sure tests pass after every commit. ✅
- [x] New/refactored code is following same conventions as the rest of the code base. 🧬
- [x] New/refactored code is tested. ⚙
- [x] Documentation has been updated 🧾

## Pre-merge checklist
- [x] Commit history is consistent and clean. 👌

### CODE REVIEW

File: docs/explorer.rst [hunk 1]
```diff
@@ -255,16 +255,16 @@ the ``has`` filter to find cases that have ``4d-seismic`` data:
 
 .. code-block:: python
 
-    from fmu.sumo.explorer import Explorer, Filters
+    from fmu.sumo.explorer import Explorer, filters
```
Comment (rwiker) [inline docs/explorer.rst:258]: Why is `Filters` changed to `filters` here?

**Review by rwiker — DISMISSED** [review]
LGTM, but the change from `Filters` to `filters` in `docs/explorer.rst` are wrong, and should be reverted.

File: docs/explorer.rst (hunk 1 above)
Comment (equinor-ruaj) [reply docs/explorer.rst:258]: I guess the convention is lowercase filenames? Which `Filters.py` is breaking.

Should `Filters.py` be renamed to `filters.py` as well?

Comment (sean-sinclair) [reply docs/explorer.rst:258]: Oops. I thought I had renamed the file to filters.py.

**Comment by sean-sinclair** [comment]
> LGTM, but the change from `Filters` to `filters` in `docs/explorer.rst` are wrong, and should be reverted.

pep-8 recommends the use of `snake_case` for module names. 

Since the naming of all the other files in the same folder is using snake_case I'll maintain that convention by renaming the file to filters.py

**Review by equinor-ruaj — APPROVED** [review]
👍

**Pull request merged**